File: leetCode/dayLeetCode/envelopes/longFindLongth.js


//#region 
/**
 * @param {number[][]} envelopes
 * @return {number}
 */

envelopes = [[5, 4], [6, 4], [6, 7], [2, 3]];

var maxEnvelopes = function(envelopes) {
  let arr = envelopes.sort((x,y) => {
    return x[0]-y[0]
  });
  let re = arr.map((currentArr) => {
    return currentArr;
  });
};



let LongthArr = function (array) {
  let lengthResult = [];
  for (let i = 0;i < array.length;i++) {
    let resArr = [];
    const firstValue = array[i];
    resArr.push(firstValue);
    for (let i2 = i + 1;i2 < array.length;i2++) {
      // 使用双指针
      let frontValue = firstValue;
      let endValue = array[i2];
      if (frontValue <= endValue) {
        resArr.push(endValue);
        // 双指针都向前推进
        frontValue = endValue;
        endValue = array[++i2];
      } else {
        // 双指针都向前推进
        frontValue = endValue;
        endValue = array[++i2];
      }
    }
    // 取出每一个递增数组的长度并放入到长度数组中去
    lengthResult.push(resArr.length)
  }
  let maxValue = lengthResult[0]
  for (let i = 1;i < lengthResult.length;i++) {
    let nextValue = lengthResult[i];
    maxValue = maxValue > nextValue ? maxValue : nextValue;
  }
  return maxValue;
}
let e = [1, 2, 3, 4, 5, 8, 2, 3, 4, 5, 6, 7];
let out = LongthArr(e);
console.log(out)


/**
 * @param {number[]} nums
 * @return {number}
 */

/* 
var findLengthOfLCIS = function(array) {
    let lengthResult = [];
    for (let i = 0;i < array.length;i++) {
        let resArr = [];
        const firstValue = array[i];
        resArr.push(firstValue);

        let frontValue = firstValue;
        let endValue;
        for (let i2 = i + 1,endValue = array[i2];i2 < array.length;) {
            // 使用双指针
            
            if (frontValue < endValue) {
                resArr.push(endValue);
                // 双指针都向前推进
                frontValue = endValue;
                endValue = array[++i2];
            } else {
                // 双指针都向前推进
                frontValue = endValue;
                endValue = array[++i2];
            }
        }
        // 取出每一个递增数组的长度并放入到长度数组中去
        lengthResult.push(resArr.length)
  }
  let maxValue = lengthResult[0]
  for (let i = 1;i < lengthResult.length;i++) {
    let nextValue = lengthResult[i];
    maxValue = maxValue > nextValue ? maxValue : nextValue;
  }
  return maxValue;
};
 */
//#endregion

let FindLongth = function (numArr) {
  // 这里是保存每次遍历数组保存的结果，如果需要节约内存，可以直接使用numArr前面的部分存储结果值，
  //但是本着纯函数的原则，还是算了吧
  if (0===numArr.length) {
    return 0;
  }
  let start = -1;
  let max = 0;
  for (let i = 0; i < numArr.length; i++) {
    if (numArr[i]<numArr[1+i]) {
      continue;
    } else {
      max = i - start > max ? i - start : max;
      start = i;
    }
  }
  return max;
}
// 龙哥写的求最长子序列

File: leetCode/dayLeetCode/envelopes/longFindLongth.c
/* 给定一个未经排序的整数数组，找到最长且 连续递增的子序列，并返回该序列的长度。
  连续递增的子序列 可以由两个下标 l 和 r（l < r）确定，如果对于每个 l <= i < r，
  都有 nums[i] < nums[i + 1] ，那么子序列[nums[l], nums[l + 1], ..., nums[r - 1], nums[r]] 就是连续递增子序列。
来源：力扣（LeetCode）
链接：https : //leetcode-cn.com/problems/longest-continuous-increasing-subsequence
著作权归领扣网络所有。商业转载请联系官方授权，非商业转载请注明出处。
 */
int findLengthOfLCIS(int *nums, int numsSize)
{
  if (0 == numsSize)
  {
    return 0;
  }
  int start = -1;
  int max = 0;
  int tags = 0;
  for (int i = 0; i < numsSize; ++i)
  {
    if (numsSize - 1 != i && nums[i] < nums[1 + i])
    {
      ++tags;
      max = i - start > max ? i - start : max;
      continue;
    }
    else
    {
      max = i - start > max ? i - start : max;
      start = i;
    }
  }
  if (numsSize - 1 == tags)
  {
    return tags + 1;
  }
  return max;
}